src/sqlancer/stonedb/gen/StoneDBExpressionGenerator.java

```diff
@@ -0,0 +1,276 @@
+package sqlancer.stonedb.gen;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import sqlancer.IgnoreMeException;
+import sqlancer.Randomly;
+import sqlancer.common.ast.BinaryOperatorNode.Operator;
+import sqlancer.common.ast.newast.NewBetweenOperatorNode;
+import sqlancer.common.ast.newast.NewBinaryOperatorNode;
+import sqlancer.common.ast.newast.NewCaseOperatorNode;
+import sqlancer.common.ast.newast.NewFunctionNode;
+import sqlancer.common.ast.newast.NewInOperatorNode;
+import sqlancer.common.ast.newast.NewUnaryPostfixOperatorNode;
+import sqlancer.common.ast.newast.NewUnaryPrefixOperatorNode;
+import sqlancer.common.ast.newast.Node;
+import sqlancer.common.gen.UntypedExpressionGenerator;
+import sqlancer.stonedb.StoneDBProvider.StoneDBGlobalState;
+import sqlancer.stonedb.StoneDBSchema.StoneDBColumn;
+import sqlancer.stonedb.StoneDBSchema.StoneDBDataType;
+import sqlancer.stonedb.ast.StoneDBConstant;
+import sqlancer.stonedb.ast.StoneDBExpression;
+
+public class StoneDBExpressionGenerator extends UntypedExpressionGenerator<Node<StoneDBExpression>, StoneDBColumn> {
+
+    private final StoneDBGlobalState globalState;
+
+    public StoneDBExpressionGenerator(StoneDBGlobalState globalState) {
+        this.globalState = globalState;
+    }
+
+    private enum Expression {
+        UNARY_PREFIX, UNARY_POSTFIX, BINARY_COMPARISON, BINARY_LOGICAL, BINARY_ARITHMETIC, BINARY_BITWISE, BETWEEN, IN,
+        CASE
+    }
+
+    @Override
+    public Node<StoneDBExpression> negatePredicate(Node<StoneDBExpression> predicate) {
+        return null;
+    }
+
+    @Override
+    public Node<StoneDBExpression> isNull(Node<StoneDBExpression> expr) {
+        return null;
+    }
+
+    @Override
+    public Node<StoneDBExpression> generateConstant() {
+        if (Randomly.getBooleanWithSmallProbability()) {
+            return StoneDBConstant.createNullConstant();
+        }
+        StoneDBDataType type = StoneDBDataType.getRandomWithoutNull();
+        switch (type) {
+        case INT:
+            return StoneDBConstant
+                    .createIntConstant(globalState.getRandomly().getInteger(Integer.MIN_VALUE + 1, Integer.MAX_VALUE));
```

Comment:
I assume it's deliberate that we are not generating `Integer.MIN_VALUE` here?

Reply:
I original thought the "For INT type: -2147483648 is reserved to indicate INT_NULL in Tianmu engine, Minimum Value Signed start from -2147483647." So we need to start from `Integer.MIN_VALUE + 1`

I thought originally that we should generate valid non value. But it seams that we call generate null value also?